common/rdr/FdOutStream.cxx

```diff
@@ -193,7 +194,14 @@ int FdOutStream::writeWithTimeout(const void* data, int length, int timeoutms)
     return 0;
 
   do {
+    // select only guarantees that you can write SO_SNDLOWAT without
+    // blocking, which is normally 1. Use MSG_DONTWAIT to avoid
+    // blocking, when possible.
+#ifdef WIN32
     n = ::write(fd, data, length);
+#else
+    n = ::send(fd, data, length, MSG_DONTWAIT);
+#endif
```

Comment:
Can't we just get rid of the WIN32 hack instead and call `send()` everywhere?

Reply:
WIN32 does not support MSG_DONTWAIT, but if you prefer, I can change to:
```
#ifdef WIN32
    n = ::send(fd, data, length, 0);
#else
    n = ::send(fd, data, length, MSG_DONTWAIT);
#endif
```

Reply:
Well that's annoying. But yes, the alternative would be better. But I think you should have `#ifndef MSG_DONTWAIT` rather than assuming a direct correlation with `WIN32`.

Reply:
Can do, but also note that send() on Windows takes a "const char*" rather than a "void char*", so in that case, it will be:
```
#ifndef MSG_DONTWAIT
    n = ::send(fd, (const char*)data, length, 0);
#else
    n = ::send(fd, (const char*)data, length, MSG_DONTWAIT);
#endif
```
Do you like that?